### PR TITLE
Remove selected store setting

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part 1 from: #5927 

Part 2 is not urgent to merge and will be merged on `trunk` here #5939

### Description
After enabling the new more menu tab, users can see store details and switch store directly from the more menu. So in order to avoid any possible confusion, we are removing the setting with the site details. This should be released in `release/8.6` so after merging this PR we should generate new betas versions. cc: @jkmassel 

### Testing instructions
Open settings (from More Menu now) and check that row with the site details is not visible anymore: 

| Before  | After|
| ------------- | ------------- |
|  |  |

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by hafizrahman — APPROVED** [review]
Changes LGTM. Just left a question but otherwise should be OK to merge.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt [hunk 1]
```diff
@@ -173,27 +170,6 @@ class MainSettingsFragment : Fragment(R.layout.fragment_settings_main), MainSett
             findNavController().navigateSafely(R.id.action_mainSettingsFragment_to_licensesFragment)
         }
 
-        /*
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt:176]: Just curious, I wonder if the removal of the comment here would be better as its own commit so it's easier to revert without touching the other changes in the same commit?

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt:176]: Good point. I believe all the changes should go together in one commit. Because in case we want to revert this to re-enable the settings options, we need to revert the xml file changes too, as well as the  removed `updateStore()` function. Having it in separate commits may lead to forgetting to revert needed changes other than the commit.

**Review by malinajirka — COMMENTED** [review]
Thanks so much for working on this!

I noticed that there are more parts of the code which I believe we can remove now
1. [`siteChanged` property from AppSettingsFragment](https://github.com/woocommerce/woocommerce-android/blob/366e5d9f3c8b72583b7f6bff277e0adbff726818/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt#L47)
2. [onSiteChanged() method](https://github.com/woocommerce/woocommerce-android/blob/366e5d9f3c8b72583b7f6bff277e0adbff726818/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt#L109)
3. [OnSiteChanged definition](https://github.com/woocommerce/woocommerce-android/blob/49f3fdbbe6d04fb185be182124ed8b49a13a00b1/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt#L51)
4. [Probably whole onActivityResult()](https://github.com/woocommerce/woocommerce-android/blob/49f3fdbbe6d04fb185be182124ed8b49a13a00b1/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt#L220)
5. [Definition of RESULT_CODE_SITE_CHANGED and all its usages](https://github.com/woocommerce/woocommerce-android/blob/366e5d9f3c8b72583b7f6bff277e0adbff726818/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt#L35)
6. [MainActivity onSiteChanged handling](https://github.com/woocommerce/woocommerce-android/blob/ffa76609777910da5a6b4a4a21f0e73c930d076c/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt#L564)

Wdyt? If there is a pretty high chance we'll want to revert this, feel free to follow the original approach and remove the code when we make the final decision. However, I think it'd be good to clean up all the code which is not necessary eventually. Otherwise, it increases complexity of the codebase and might confuse us in the future.

**Comment by JorgeMucientes** [comment]
Thanks for adding details of all the additional code that can be removed @malinajirka, that is great. However I'm going to apply them in a different branch based on `trunk` along with the commits already done here. I realized that `release/8.6` will be deleted after merging it back into `trunk` and the merge commit generated will contain all the fixes applied to `release/8.6` which will make it harder in case we want to revert changes from this PR in the future. Will handle all in a different task. 

Also I wanted to mention that while looking at your suggested code removal I realized that this function: 
```kotlin 
    override fun onSiteChanged() {
        if (FeatureFlag.CARD_READER.isEnabled()) presenter.clearCardReaderData()
        siteChanged = true
        setResult(RESULT_CODE_SITE_CHANGED)

        prefs.resetSitePreferences()
    }
```
Does 2 calls that we are missing in the current implementation
```kotlin 
        if (FeatureFlag.CARD_READER.isEnabled()) presenter.clearCardReaderData()
        prefs.resetSitePreferences()
```
I think those calls should not belong there as it easy to forget to migrate them in other places were we change site. Like it just happened here. 
My proposal is to move that missing logic to: 
- MainPresenter.selectedSiteChanged()  ->     `if (FeatureFlag.CARD_READER.isEnabled()) presenter.clearCardReaderData()`
- SitePickerPresenter -> `prefs.resetSitePreferences()`

Apply those changes in this branch.
Wdyt?

**Comment by JorgeMucientes** [comment]
I've added 3 different commits applying the changes mentioned earlier: 

- Reset site preferences when changin site -> [here](https://github.com/woocommerce/woocommerce-android/pull/5941/commits/f68049a391621dada4a45e0e19f1d20206eed7f8)
- Clear card reader data. For that I created a new use case class that we then can reuse in different places -> [changes](https://github.com/woocommerce/woocommerce-android/pull/5941/commits/264c5caf1ba748d58f1b3f2740bfa0054162cf6d)
- Removed all the code suggested from @malinajirka in the previous comment -> [here](https://github.com/woocommerce/woocommerce-android/pull/5941/commits/75d8c63819c758c0bbfb99ada4b28900d9e0dd64)

If you are ok with these changes what I think we can do is merge this into `release/8.6` and then I will cherry-pick the merged commit into a `trunk` based branch and then merge that branch into `trunk`. This way we will keep a single commit in `trunk` that will easily enable us to revert this changes in case we want to add back the switch store option inside settings and remove it from more menu. 

Let me know what you think. I'm not a big fan of doing that many changes over a release branch, but I feel it was needed. Otherwise, we wouldn't be able to fix all the missing code we were running before removing the switch site out of settings.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt [hunk 2]
```diff
@@ -200,7 +199,6 @@ class MainSettingsFragment : Fragment(R.layout.fragment_settings_main), MainSett
         // knows it has changed
         if (requestCode == RequestCodes.SITE_PICKER && resultCode == Activity.RESULT_OK) {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt]: I think we can remove this whole `if` statement or even whole `onActivityResult` as `requestCode == RequestCodes.SITE_PICKER` is never true. Wdyt?

**Review by malinajirka — COMMENTED** [review]
>  I realized that release/8.6 will be deleted after merging it back into trunk and the merge commit generated will contain all the fixes applied to release/8.6 which will make it harder in case we want to revert changes from this PR in the future. 

I'm not sure I follow. AFAIK we don't squash commits before merging release branch into trunk. So trunk will imo contain all the commits from release/8.6. What might happen is that the merge commit will contain extra changes in case of a conflict - do we expect any conflicts or did you mean something completely different?

> If you are ok with these changes what I think we can do is merge this into release/8.6 and then I will cherry-pick the merged commit into a trunk based branch and then merge that branch into trunk. This way we will keep a single commit in trunk that will easily enable us to revert this changes in case we want to add back the switch store option inside settings and remove it from more menu.

Related to the above, I'm not sure I follow. Are you suggesting we squash commits in this PR before merging them into `release/8.6`? I'd personally vote against it - if we need to revert, we can revert the whole PR -> github supports that with a single button click. Having a single commit is maybe slightly easier but we lose a bit of history and is imo not worth it.

Also, I think re-applying the changes might be easier then reverting multiple PRs anyway. So I personally wouldn't worry about it too much.

I'm not done with the review as I haven't tested the app yet. But I want to post this comment before doing so to keep the conversation flowing.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/main/MainPresenterTest.kt [hunk 3]
```diff
@@ -197,6 +200,15 @@ class MainPresenterTest : BaseUnitTest() {
         }
     }
 
+    @Test
+    fun `When selected site changes, then card reader data is cleared`() = testBlocking {
```
Comment (malinajirka) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/main/MainPresenterTest.kt:204]: ❤️

**Comment by JorgeMucientes** [comment]
> I'm not sure I follow. AFAIK we don't squash commits before merging release branch into trunk. So trunk will imo contain all the commits from release/8.6. What might happen is that the merge commit will contain extra changes in case of a conflict - do we expect any conflicts or did you mean something completely different?

Thanks for requesting for clarifications @malinajirka . I'll try to explain better because it is very likely I'm missing something.

 Considering this generic schema: 
Screenshot 2022-02-28 at 18 25 16

Let's assume that the green branch is `release/8.6` and A B and C are different fixes we applied over it.
-  Say we merge this PR and that results in commit **B** on the picture 
- Then we merge green branch (release/8.6) into blue on the picture
- Then **delete** the green branch 

If you want to revert commit B **only**, how would you do that? You have deleted `release/8.6` and no longer have the history of A , B and C right? Or maybe I don't understand `git merge` 😅 😬 

Considering the previous assumptions and schema, my idea was to cherry-pick commit B manually into `trunk` to make sure we have these PR changes in `trunk` history in an isolated way. Does that make sense? 

If the way we merge `release/8.6` into `trunk` is by rebasing and then fast-forward merging, then forget about everything I said 😄 . I assumed that the way of work for merging `release/8.6` would be like any other feature branch we work on.

**Comment by malinajirka** [comment]
> If you want to revert commit B only, how would you do that? You have deleted release/8.6 and no longer have the history of A , B and C right? Or maybe I don't understand git merge 😅 😬

I believe this is not the case. Merge keeps the whole history of all commits => it merges two commit histories from two branches into one history of commits (it still includes all the commits => that's why it's called merge). The merge commit is added to the final history to show how the files from both branches were actually merged.

**Comment by malinajirka** [comment]
> If you want to revert commit B only, how would you do that? You have deleted release/8.6 and no longer have the history of A , B and C right?

If this was the case, wouldn't `trunk` contain only merge commits since all the other commits wouldn't exist anymore - the PR/feature branches got deleted?

**Review by malinajirka — APPROVED** [review]
Thanks for working on this @JorgeMucientes! I've re-tested it and it seems it works as expected 🥳 !

Since we are merging this into the release branch relatively late in the release cycle, I'd appreciate if @hafizrahman could re-test it too.

@jkmassel This PR needs to go into the 8.6 but I'd prefer to wait for @hafizrahman to test the changes. If you could double check our convo about behavior of `git merge` in the meatime it'd be much appreciated. Thanks!

**Comment by JorgeMucientes** [comment]
> If this was the case, wouldn't trunk contain only merge commits since all the other commits wouldn't exist anymore - the PR/feature branches got deleted?


Wow what a facepalm to learn how merge actually works at this point in life 😅. Somehow I believed that those additional commits that appeared in the git tree next  to `trunk`  (or any branch for that matter) were always undeleted "feature" branches that had been merged into it. But yeah I see now you can delete the feature branch and the commits still there 🤦🏼 🤦🏼‍♂️  

Anyway, thanks so much for reviewing this @malinajirka and lets wait for @hafizrahman to check it out too!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt (hunk 2 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/MainSettingsFragment.kt]: Sure! Removed that and the unused method `showSitePickerForResult` that triggered that onActivityResult()

Good catch!

**Comment by malinajirka** [comment]
Perhaps, you were using squash before merge in the other companies you worked for. It's a pretty common practice and it removes the commit history of the feature branch (or to be precise, it squashes it into a single commit). Anyway, glad we are on the same page - everyday is an opportunity to learn something new :)!

I just noticed @hafizrahman is AFK => I retested it one more time and it LGTM. Merging now :shipit:
